### PR TITLE
toolchain/ghs: Fix green hills toolchain build Vela  "no storage class…

### PR DESCRIPTION
**Pull request by extinguish**
## Summary

Fix green hills toolchain build Vela  "no storage class or type specifier" error

### CODE REVIEW

File: libs/libxx/libcxxmini/libxx_cxa_guard.cxx [hunk 1]
```diff
@@ -44,7 +44,7 @@ typedef int __guard;
 // variable and the first byte contains 0 prior to initialization, and
 // 1 after.
 
-__extension__ typedef int __guard __attribute__((mode(__DI__)));
```
Comment (anchao) [inline libs/libxx/libcxxmini/libxx_cxa_guard.cxx:47]: why not define to empty in compiler.h?

Comment (extinguish) [reply libs/libxx/libcxxmini/libxx_cxa_guard.cxx:47]: > why not define to empty in compiler.h?

is that mean to keep "libs/libxx/libcxxmini/libxx_cxa_guard.cxx" left unchanged, and add a following empty define in "nuttx/compiler.h":
`#  define __extension__`

Comment (anchao) [reply libs/libxx/libcxxmini/libxx_cxa_guard.cxx]: Yes, let only defined in special compilers, such as GHS

File: include/nuttx/compiler.h [hunk 2]
```diff
@@ -1110,6 +1110,14 @@
 
 #endif
 
+/* Green Hills Software definitions *****************************************/
+
+#if defined(__ghs__)
+
+#  define __extension__
```
Comment (anchao) [inline include/nuttx/compiler.h:1117]: You could merge this PR with your other compiler.h changes: https://github.com/apache/nuttx/pull/12846

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by anchao — CHANGES_REQUESTED** [review]
please define ghs before gnuc and merge this PR with your other compiler.h changes: #12846

**Pull request closed without merging**